### PR TITLE
plasma-theme-switcher: init at 0.1

### PR DESCRIPTION
**Pull request by Unkn0wnCat**
###### Motivation for this change

The package "plasma-theme-switcher" allows for easily modifying the Plasma desktop theme. This is useful for use with NixOS as color schemes or themes can be set from scripts in the global configuration.

###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [x] ~~Tested, as applicable:~~ Not testable w/o full Plasma environment.
  - ~~[NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))~~
  - ~~and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)~~
  - ~~or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)~~
  - ~~made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages~~
- [x] ~~Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)~~ Nothing depends on this.
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/top-level/all-packages.nix
Comment (ajs124) [inline pkgs/top-level/all-packages.nix:34560]: we don't just add to the end of `all-packages.nix`

also, this could/should use `libsForQt5.callPackage`

File: pkgs/applications/misc/plasma-theme-switcher/default.nix
Comment (ajs124) [inline pkgs/applications/misc/plasma-theme-switcher/default.nix:38]: `gpl2` isn't a precise license. please choose `gpl2Only` or `gpl2Plus`

File: pkgs/applications/misc/plasma-theme-switcher/default.nix
Comment (ajs124) [inline pkgs/applications/misc/plasma-theme-switcher/default.nix:24]: can be omitted, just add `cmake` and `extra-cmake-modules` to `nativeBuildInputs` (and remove from `buildInputs`)

File: pkgs/applications/misc/plasma-theme-switcher/default.nix
Comment (ajs124) [inline pkgs/applications/misc/plasma-theme-switcher/default.nix:2]: we don't pass in `pkgs`, like this. please specify your actual dependencies here. e.g.
```
, cmake
, extra-cmake-modules
, qtbase
, kdeFrameworks
, wrapQtAppsHook
```

File: pkgs/applications/misc/plasma-theme-switcher/default.nix
Comment (ajs124) [inline pkgs/applications/misc/plasma-theme-switcher/default.nix:7]: version should be without `v` for reasons that I'd rather not go into

**Review by ajs124 — CHANGES_REQUESTED** [review]

File: pkgs/top-level/all-packages.nix
Comment (Unkn0wnCat) [reply pkgs/top-level/all-packages.nix:34560]: In what order is the package supposed to be added? I've tried finding alphabetical order first, but with no success.

Please excuse my question, this is my first time working with/on nix

Comment (ajs124) [reply pkgs/top-level/all-packages.nix:34560]: There's a comment at the top of the file trying to explain how it's supposed to be sorted, but it's quite a mess, tbh

Comment (Unkn0wnCat) [reply pkgs/top-level/all-packages.nix]: I've tried to sort it in near a similar application now, I honestly missed the comment first time, sorry for that.

Let me know if the new sorting works! :)

File: maintainers/maintainer-list.nix
Comment (IvarWithoutBones) [inline maintainers/maintainer-list.nix:6335]: This should be in a seperate commit

File: pkgs/applications/misc/plasma-theme-switcher/default.nix
Comment (IvarWithoutBones) [inline pkgs/applications/misc/plasma-theme-switcher/default.nix:24]: `DCMAKE_BUILD_TYPE` is implicitly set by cmake already. Also, I dont think the `DCMAKE_INSTALL_PREFIX` option does anything, since `installPhase` is overwritten.
```suggestion
```

File: pkgs/applications/misc/plasma-theme-switcher/default.nix
Comment (IvarWithoutBones) [inline pkgs/applications/misc/plasma-theme-switcher/default.nix:28]:
```suggestion
    runHook preInstall

    mkdir -p $out/bin
    install -Dm755 ./plasma-theme $out/bin/plasma-theme
    
    runHook postInstall
```

Also, is there a reason why we can't use cmake's install procedure?

File: pkgs/applications/misc/plasma-theme-switcher/default.nix
Comment (IvarWithoutBones) [inline pkgs/applications/misc/plasma-theme-switcher/default.nix:35]:
```suggestion
  meta = with lib; {
    homepage = "https://github.com/maldoinc/plasma-theme-switcher/";
    description = "A KDE Plasma theme switcher";
    license = with licenses; [ gpl2Only ];
    maintainers = with maintainers; [ kevink ];
```

**Review by IvarWithoutBones — COMMENTED** [review]
I have a few suggestions :)

File: maintainers/maintainer-list.nix
Comment (Unkn0wnCat) [reply maintainers/maintainer-list.nix:6335]: Oh I'll try splitting it into two tomorrow, but this will mean this PR will be borked until the merge of the maintainer PR is through... It's kind of a catch 22

File: pkgs/applications/misc/plasma-theme-switcher/default.nix
Comment (Unkn0wnCat) [reply pkgs/applications/misc/plasma-theme-switcher/default.nix:28]: The source code does not provide an install target. Leaving the override out will cause the installation to fail on this package

File: maintainers/maintainer-list.nix
Comment (IvarWithoutBones) [reply maintainers/maintainer-list.nix:6335]: You can do it in this PR, that's not an issue

Comment (Unkn0wnCat) [reply maintainers/maintainer-list.nix:6335]: Ah sorry, I misread commit for PR, my bad

File: pkgs/applications/misc/plasma-theme-switcher/default.nix
Comment (Unkn0wnCat) [reply pkgs/applications/misc/plasma-theme-switcher/default.nix:35]: Good call, I will apply this tomorrow!

File: maintainers/maintainer-list.nix
Comment (IvarWithoutBones) [reply maintainers/maintainer-list.nix:6335]: No worries :)

File: pkgs/applications/misc/plasma-theme-switcher/default.nix
Comment (Unkn0wnCat) [reply pkgs/applications/misc/plasma-theme-switcher/default.nix:24]: Will test this tomorrow, to be honest I just looked at what other packages were doing - first time creating a package :p

File: maintainers/maintainer-list.nix
Comment (Unkn0wnCat) [reply maintainers/maintainer-list.nix:6335]: It's split now! :)

File: pkgs/applications/misc/plasma-theme-switcher/default.nix
Comment (IvarWithoutBones) [inline pkgs/applications/misc/plasma-theme-switcher/default.nix:8]: This is only used with `wrapQtAppsHook`, i think it can be removed
```suggestion
```

Comment (Unkn0wnCat) [reply pkgs/applications/misc/plasma-theme-switcher/default.nix:8]: It refuses to compile without that line.

`Error: wrapQtAppsHook is not used, and dontWrapQtApps is not set.` in the qtPreHook

Comment (IvarWithoutBones) [reply pkgs/applications/misc/plasma-theme-switcher/default.nix:8]: Huh, i wasn't aware that's a thing. Thanks for letting me know :)

**Review by IvarWithoutBones — APPROVED** [review]
Binary seems to run, and the code LGTM :+1: 

Result of `nixpkgs-review pr 160183` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>plasma-theme-switcher</li>
  </ul>
</details>

**Comment by Unkn0wnCat** [comment]
Thanks @ajs124 and @IvarWithoutBones for helping me with this! I've learned a lot about Nix and NixOS in the process! 🙂

**Comment by IvarWithoutBones** [comment]
Happy to help :blush:

**Pull request merged**
